### PR TITLE
Stronger email checks for analytics

### PR DESCRIPTION
**Pull request by benrudolph**
@snopoke @emord went back and forth on the best way to do this, but finally settled on this. http://manage.dimagi.com/default.asp?242172 for more context. simon, i didn't use `notify_error` cause i thought it'd be too noisy and not that actionable. let me know if you think i should still use that

### CODE REVIEW

File: corehq/apps/analytics/tasks.py [hunk 1]
```diff
@@ -368,7 +369,8 @@ def track_periodic_data():
     submit = []
     for user in users_to_domains:
         email = user.get('email')
-        if not email:
+        if not email or not validate_email(email, verify=True):
```
Comment (benrudolph) [inline corehq/apps/analytics/tasks.py]: One thing to note about `validate_email` is that it returns `None` if it's unsure. so this is being extra stringent. We can change this so it only skips when we now it's not valid

Comment (benrudolph) [inline corehq/apps/analytics/tasks.py]: there's also this email lib that checks the domain name: https://github.com/JoshData/python-email-validator. overall it seems like a better library. i might go with that

**Comment by emord** [comment]
Curious why you think this will be too noisy? I assume that having an invalid email after registering would not be that common,

**Comment by benrudolph** [comment]
we save the user account even if the email account hasn't been validated yet

File: corehq/apps/analytics/tasks.py (hunk 1 above)
Comment (snopoke) [inline corehq/apps/analytics/tasks.py]: We may also want to make this configurable for the sake of NIC (which may not have good internet access to do the validation).

Comment (emord) [inline corehq/apps/analytics/tasks.py]: This isn't being run on nic though since it's submitting to hubspot right?

Comment (benrudolph) [inline corehq/apps/analytics/tasks.py]: i think it still gets run (unless there is something excluding it from being picked up as a task), but you're right i doubt anything works so it's probably not a big deal

File: requirements/requirements.txt [hunk 2]
```diff
@@ -95,3 +95,5 @@ github3.py==1.0.0a4
 django-bulk-update==1.1.10
 csiphash==0.0.4
 json-delta==2.0
+email_validator==1.0.1
```
Comment (nickpell) [inline requirements/requirements.txt]: @benrudolph Any reason you went with this over the built in django validator? https://docs.djangoproject.com/en/1.8/ref/validators/#validate-email

Comment (benrudolph) [inline requirements/requirements.txt]: cause it checks if the email is deliverable (important for hubspot): https://github.com/JoshData/python-email-validator#overview. it's easy to have a fake email that passes the regex but isn't an actual account

Comment (nickpell) [inline requirements/requirements.txt]: ooooh very cool

**Pull request merged**